### PR TITLE
Projection EDFM

### PR DESCRIPTION
**Pull request by CusiniM**
In this PR we implement a first version of the Projection EDFM to account for impermeable embedded fractures/faults.

Main changes:
- Slightly refactored the EDFM-part of TwoPointFluxApproximation to split M-M and F-M connection computation;
- Added a ProjectionEDFMHelper that does all the pEDFM stuff;
- Added a pEDFM sinplePhase integratedTest;
- Added a pEDFM compositional multiphase integratedTest;

What works:
- The code has all the face selection, geometric proccessing, and transmissibility calculation

What's missing:
- No full projection, but assume that it occupies all the selected faces.
- No fracture-fracture intersections

Note: we could probably make this `ProjectionEDFMHelper` just a namespace since it has no data. 

Rebaseline: https://github.com/GEOSX/integratedTests/pull/150

### CODE REVIEW

**Review by francoishamon — APPROVED** [review]
Looks great!

**Review by klevzoff — APPROVED** [review]
Some more minor comments

File: src/coreComponents/finiteVolume/ProjectionEDFMHelper.hpp [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+                        EmbeddedSurfaceToCellStencil & edfmStencil,
+                        string const & embeddedSurfaceRegionName );
+
+  /*
+   * @brief add Fracture-matrix connections to the edfmStencil
+   * and remove the appropriate connections from the cellStencil
+   */
+  void addNonNeighboringConnections() const;
+
+  virtual ~ProjectionEDFMHelper() = default;
```
[the remark this answers is not in the document]
Comment (CusiniM) [reply src/coreComponents/finiteVolume/ProjectionEDFMHelper.hpp:68]: I would like to make this a `namespace` with only free functions.

File: src/coreComponents/finiteVolume/ProjectionEDFMHelper.cpp [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+  m_edfmStencil.add( numElems,
+                     stencilCellsRegionIndex.data(),
+                     stencilCellsSubRegionIndex.data(),
+                     stencilCellsIndex.data(),
+                     stencilWeights.data(),
+                     m_edfmStencil.size() );
+}
+
+ void ProjectionEDFMHelper::
+  fractureMatrixTransmissilibility( CellDescriptor const & neighborCell,
```
[the remark this answers is not in the document]
Comment (CusiniM) [reply src/coreComponents/finiteVolume/ProjectionEDFMHelper.cpp:237]: it actually only computes the geometrical part of the transmissiblity.

File: src/coreComponents/finiteVolume/ProjectionEDFMHelper.cpp [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+
+        // zero out matrix-matrix connections that are replaced by fracture-matrix connections
+        // I assume that the m-m connection index is equal to the face id
+        m_cellStencil.zero( faceIdx );
+      }
+    }
+  }
+}
+
+std::list< localIndex > ProjectionEDFMHelper::selectFaces( FixedOneToManyRelation const & subRegionFaces,
```
[the remark this answers is not in the document]
Comment (CusiniM) [reply src/coreComponents/finiteVolume/ProjectionEDFMHelper.cpp]: done

File: src/coreComponents/mainInterface/ProblemManager.cpp [hunk 4]
```diff
@@ -723,6 +723,11 @@ DomainPartition const & ProblemManager::getDomainPartition() const
   return getGroup< DomainPartition >( keys::domain );
 }
 
+GeometricObjectManager const & ProblemManager::getGeometricObjectManager() const
+{
+  return getGroup< GeometricObjectManager >( groupKeys.geometricObjectManager );
+}
+
```
[the remark this answers is not in the document]
Comment (CusiniM) [reply src/coreComponents/mainInterface/ProblemManager.cpp:730]: no it's not and I think this is a `singleton` now so there is no need for this and there is no need to use `getGroup` to grab it.

**Comment by CusiniM** [comment]
I think this can be merged.

**Pull request merged**